12-offer-encoding.md

```diff
@@ -0,0 +1,1057 @@
+      - if it includes `recurrence_paywindow`:
+        - MUST set `seconds_before` to the maximum number of seconds prior to
+          a period for which it will accept payment or invoice_request for that period.
+        - MUST set `seconds_after` to the maximum number of seconds into to a
+          period for which it will accept payment or invoice_request for that period.
+        - MAY NOT enforce this for the initial period for offers without `recurrence_base`
+        - SHOULD NOT set `seconds_after` to greater than the maximum number of
+          seconds in a period.
+        - if it `amount` is specified and the node will proportionally reduce
+          the amount charged for a period payed after the start of the period:
+          - MUST set `proportional_amount` to 1 
+        - otherwise:
+          - MUST set `proportional_amount` to 0
+  - otherwise:
+    - MUST NOT include `recurrence_base`.
+    - MUST NOT include `recurrence_paywindow`.
+    - MUST NOT include `recurrence_limit`.
+  - if `send_invoice` is present:
+    - if the offer is for a partial or full refund for a previously-paid
+      invoice:
+      - SHOULD set `refunded_payment_hash` to the `payment_hash` of that
+        invoice.
+  - otherwise:
+    - MUST NOT set `refunded_payment_hash`.
+
+A reader of an offer:
+  - if `features` contains unknown _odd_ bits that are non-zero:
+    - MUST ignore the bit.
+  - if `features` contains unknown _even_ bits that are non-zero:
+    - MUST NOT respond to the offer.
+    - SHOULD indicate the unknown bit to the user.
+  - if `node_id` or `description` is not set:
+    - MUST NOT respond to the offer.
+  - if `signature` is present, but is not a valid signature using `node_id` as described in [Signature Calculation](#signature-calculation):
+    - MUST NOT respond to the offer.
+  - SHOULD gain user consent for recurring payments.
+  - SHOULD allow user to view and cancel recurring payments.
+  - if it uses `amount` to provide the user with a cost estimate:
+    - MUST warn user if amount of actual invoice differs significantly
+        from that expectation.
+  - SHOULD not respond to an offer if the current time is after
+    `absolute_expiry`.
+  - FIXME: more!
+
+## Rationale
+
+It's quite reasonable to set a `recurrence_paywindow` with seconds_after
+equal to 0, but obviously this should not apply to the initial period if
+there is no recurrence_base.
+
+A signature is optional, because it makes for a longer string (potentially
+limiting QR code use on low-end cameras); if the offer has an error, no
+invoice will be given (or, for `send_invoice` offers, accepted), since
+the `offer_id` already covers all the non-signature fields.
+
+# Invoice Requests
+
+Invoice Requests are a request for an invoice; the human-readable prefix for
+invoices is `lnr`.
+
+## TLV Fields for `invoice_request`
+
+1. `tlv_stream`: `invoice_request`
+2. types:
+    1. type: 2 (`chains`)
+    2. data:
+        * [`...*chain_hash`:`chains`]
```

Comment:
In what case would we need to send multiple chain hashes? And if we do, what do we expect? One invoice per chain?

Reply:
No, you get one invoice.  invoice_request chains needs to be a non-strict subset of offer chains, and invoice chains needs to be a non-strict subset of invoice_request chains.

In practice, I expect only one chain here.